pydantic_core/core_schema.py

```diff
@@ -1492,17 +1598,63 @@ def fn(v: Any, info: core_schema.ValidationInfo) -> str:
     return dict_not_none(
         type='function',
         mode='before',
-        function=function,
+        function={'type': 'function', 'call': function},
         schema=schema,
         ref=ref,
         metadata=metadata,
         serialization=serialization,
     )
 
 
-def function_after_schema(
+def model_field_function_after_schema(
+    function: ModelFieldValidatorFunction,
     schema: CoreSchema,
+    *,
+    ref: str | None = None,
+    metadata: Any = None,
+    serialization: SerSchema | None = None,
+) -> FunctionSchema:
+    """
+    Returns a schema that calls a validator function after validating the provided schema, e.g.:
+
+    ```py
+    from pydantic_core import SchemaValidator, core_schema
+
+    def fn(v: str, info: core_schema.ModelFieldValidationInfo) -> str:
+        assert info.data is not None
+        assert info.field_name is not None
+        return v + 'world'
+
+    func_schema = core_schema.model_field_function_after_schema(function=fn, schema=core_schema.str_schema())
+    schema = core_schema.typed_dict_schema(
+        {'a': core_schema.typed_dict_field(func_schema)}
+    )
+
+    v = SchemaValidator(schema)
+    assert v.validate_python({"a": b"hello "}) == {"a": "hello world"}
```

Comment:
at the risk of being pedantic
```suggestion
    assert v.validate_python({'a': b'hello '}) == {'a': 'hello world'}
```

More broardly, I'd be happy to give up the fight, and switch to double quotes, but we should be consistent.

Reply:
Soft +1 for switching to double quotes. Muscle memory means I still get caught by ruff daily, and I think black will convert to double quote automatically but not single. Agreed on consistency being more important.

Reply:
I would also be in favor of double quotes. But I'll swap these now and we can discuss later. And in my defense , I got confused and thought all of these docstrings were intended to be double quotes: https://github.com/pydantic/pydantic-core/blob/5ca4db70a22d9f62ce1025dde81923d87b537083/pydantic_core/core_schema.py#L1484

Reply:
Some other spots: https://github.com/pydantic/pydantic-core/blob/5ca4db70a22d9f62ce1025dde81923d87b537083/pydantic_core/core_schema.py#L2417

Reply:
Solving for now let's continue any discussion about single vs. double quotes on slack